scripts/static-vdis

```diff
@@ -22,27 +22,21 @@ def call_volume_plugin(name, command, args):
         cmd_args.extend(args)
     # on Python >= 3.3 a timout can be set, not on 2.7
     # when porting please add a timeout
-    output = subprocess.check_output(cmd_args)
+    output = subprocess.check_output(cmd_args, universal_newlines=True)
     return json.loads(output)
 
 def call_datapath_plugin(name, command, args):
     args = [ xapi_storage_script + "/datapath/" + name + "/" + command, "static-vdis" ] + args
-    output = subprocess.check_output(args)
+    output = subprocess.check_output(args, universal_newlines=True)
     return json.loads(output)
-
+        
 def read_whole_file(filename):
-    f = open(filename)
-    try:
-        return reduce(lambda x, y: x + y.decode('utf8'), f.readlines(), "").strip()
-    finally:
-        f.close()
-
+    with open(filename, 'r', encoding='utf-8') as f:
+        return ''.join(map(str.strip, f.readlines()))
```

Comment:
There are 2 bugs - and here is the exhibit which shows it:
```py
cat >test-pr5295.py <<END
import sys

if sys.version_info > (3,0):
    def read_whole_file(filename):
        with open(filename, 'r', encoding='utf-8') as f:
            return ''.join(map(str.strip, f.readlines()))
            #return ''.join(f.readlines()).strip()
else:
    def read_whole_file(filename):
        f = open(filename)
        try:
            return reduce(lambda x, y: x + y.decode('utf8'), f.readlines(), "").strip()
        finally:
            f.close()

with open(__file__ + "%s.out" % sys.version_info.major, "w") as o:
    o.write(read_whole_file(__file__))
END
python2 test-pr5295.py
python3 test-pr5295.py
diff -u test-pr5295.py[23].out
md5sum test-pr5295.py[23].out
```
For best results, paste it into and editor and run it, see below:

Output:
```py
bash test-pr5295.sh
--- test-pr5295.py2.out 2023-12-12 20:32:42.273484598 +0100
+++ test-pr5295.py3.out 2023-12-12 20:32:42.283484596 +0100
@@ -1,17 +1 @@
-import sys
-
-if sys.version_info > (3,0):
-    def read_whole_file(filename):
-        with open(filename, 'r', encoding='utf-8') as f:
-            return ''.join(map(str.strip, f.readlines()))
-            #return ''.join(f.readlines()).strip()
-else:
-    def read_whole_file(filename):
-        f = open(filename)
-        try:
-            return reduce(lambda x, y: x + y.decode('utf8'), f.readlines(), "").strip()
-        finally:
-            f.close()
-
-with open(__file__ + "%s.out" % sys.version_info.major, "w") as o:
-    o.write(read_whole_file(__file__))
\ No newline at end of file
+import sysif sys.version_info > (3,0):def read_whole_file(filename):with open(filename, 'r', encoding='utf-8') as f:return ''.join(map(str.strip, f.readlines()))#return ''.join(f.readlines()).strip()else:def read_whole_file(filename):f = open(filename)try:return reduce(lambda x, y: x + y.decode('utf8'), f.readlines(), "").strip()finally:f.close()with open(__file__ + "%s.out" % sys.version_info.major, "w") as o:o.write(read_whole_file(__file__))
\ No newline at end of file
f54a639ff4bcd8c662bc35be4def7493  test-pr5295.py2.out
aafbc3b6bce179b66d14cbc75d1b8c12  test-pr5295.py3.out
```
Expected: No output and identical md5sum for both:
```py
bash test-pr5295.sh
c389102aa49e776235c234768286658c  test-pr5295.py2.out
c389102aa49e776235c234768286658c  test-pr5295.py3.out
```
I get this when moving the hash '#' to the other line in the new function of the sample test.

```suggestion
    with open(filename, 'r', encoding='utf-8') as f:
        return ''.join(f.readlines()).strip()
```
Your change removes the newlines and all leading and trailing space from each line, but the original code does not do that as you see: strip() only used once at the end, once, not for each line!

To make sure this is not overlooked, I'm requesting this fix as a requested change, and I'll be on PTO until January, so I might not be able to dismiss the status of requested change until January 2.

This would have nearly escaped my review, and it would be much easier to deal with and find such issues if you would not put everything into one large pull request. In the future, make small PRs, one for each file, for example.

Reply:
Thank you Bernhard for the review and test.
Code updated.
Totally agree on small PRs for better review.